### PR TITLE
AND-1232 - Response card support

### PR DESCRIPTION
**Pull request by smpete**
Ready for full review

### CODE REVIEW

File: layer-atlas/src/main/java/com/layer/ui/message/choice/ChoiceButtonSet.java [hunk 1]
```diff
@@ -84,10 +93,19 @@ public void addOrUpdateChoice(final ChoiceModel choice) {
 
         choiceButton.setText(choice.getText());
 
+        // TODO Should we forgo this and just set the choice as the tag? What happens if the text changes?
```
Comment (thecombatwombat) [inline layer-atlas/src/main/java/com/layer/ui/message/choice/ChoiceButtonSet.java]: Not a bad idea... need to ensure the equals for the `ChoiceMetadata` checks for the id...

Comment (smpete) [reply layer-atlas/src/main/java/com/layer/ui/message/choice/ChoiceButtonSet.java]: Do we even need to do that or is reference equality enough here?

File: layer-atlas/src/main/java/com/layer/ui/message/choice/ChoiceMessageModel.java [hunk 2]
```diff
@@ -54,9 +51,7 @@ protected void parse(@NonNull MessagePart messagePart) {
         if (messagePart.equals(getRootMessagePart())) {
             JsonReader reader = new JsonReader(new InputStreamReader(messagePart.getDataStream()));
             mMetadata = mGson.fromJson(reader, ChoiceMessageMetadata.class);
-            for (ChoiceModel choiceModel : mMetadata.getChoices()) {
-                choiceModel.setMessage(getMessage());
-            }
+            mNodeId = MessagePartUtils.getNodeId(messagePart);
```
Comment (thecombatwombat) [inline layer-atlas/src/main/java/com/layer/ui/message/choice/ChoiceMessageModel.java]: Take a look at `MessageModel#processChildParts`... that already fetches the rootNodeId... might be worth just storing that as part of the state in `MessageModel` and adding a protected accessor to it.

File: layer-atlas/src/main/java/com/layer/ui/repository/MessageSenderRepository.java [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+        mLayerClient = layerClient;
+        mIdentityFormatter = new IdentityFormatterImpl(context);
+    }
+
+    public boolean sendChoiceResponse(ChoiceResponse choiceResponse) {
+        ResponseSender responseSender = new ResponseSender(mContext, mLayerClient);
+        String userName = mIdentityFormatter.getDisplayName(mLayerClient.getAuthenticatedUser());
+        String statusText = mContext.getString(R.string.response_message_status_text, userName, choiceResponse.getChoiceText(), "TODO Label name");
+
+        Log.d("ZZZ: nodeId: " + choiceResponse.getNodeIdToRespondTo());
```
Comment (thecombatwombat) [inline layer-atlas/src/main/java/com/layer/ui/repository/MessageSenderRepository.java]: Why the ZZZ?

File: layer-atlas/src/main/java/com/layer/ui/repository/MessageSenderRepository.java [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+        Log.d("ZZZ: response name: " + choiceResponse.getResponseName());
+        Log.d("ZZZ: choice text: " + choiceResponse.getChoiceText());
+        Log.d("ZZZ: choice id: " + choiceResponse.getChoiceId());
+        Log.d("ZZZ: statusText: " + statusText);
+        return false;
+//        return responseSender.sendChoiceResponse(messageToRespondTo, nodeIdToRespondTo, choice, statusText);
```
Comment (thecombatwombat) [inline layer-atlas/src/main/java/com/layer/ui/repository/MessageSenderRepository.java]: Which of these should we be keeping?

File: layer-atlas/src/main/java/com/layer/ui/repository/MessageSenderRepository.java (hunk 3 above)
Comment (smpete) [reply layer-atlas/src/main/java/com/layer/ui/repository/MessageSenderRepository.java]: Easy to search/filter the log on when debugging

File: layer-atlas/src/main/java/com/layer/ui/repository/MessageSenderRepository.java (hunk 4 above)
Comment (smpete) [reply layer-atlas/src/main/java/com/layer/ui/repository/MessageSenderRepository.java]: Still working through this. Not quite ready for prime time

File: layer-atlas/src/main/java/com/layer/ui/message/choice/ChoiceButtonSet.java (hunk 1 above)
Comment (smpete) [reply layer-atlas/src/main/java/com/layer/ui/message/choice/ChoiceButtonSet.java]: I don't want to do this now. I'd prefer to keep a custom `equals()` out of the model POJO.

**Comment by smpete** [comment]
Ok this is ready to go.

**Pull request merged**